src/db/file/pushes.js

```diff
@@ -94,8 +94,8 @@ const cancel = async (id, logger) => {
 const canUserCancelPush = async (id, user) => {
   return new Promise(async (resolve) => {
     const pushDetail = await getPush(id);
-    const repoName = pushDetail.repoName.replace('.git', '');
-    const isAllowed = await repo.isUserPushAllowed(repoName, user);
+    const repoUrl = pushDetail.repo.url;
```

Comment:
What about pre-existing repositories using the `repoName` property in the database? We need to think about how the update propagates to the previous data model and its impact on already stored properties in DB.